torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -130,7 +131,7 @@ struct TORCH_API AccumulateGrad : public Node {
             new_grad.sizes(),
             new_grad.options()));
       } else {
-        if (new_grad.is_sparse()) {
+        if (new_grad.is_sparse() || new_grad.is_mkldnn()) {
```

Comment:
Why is mkldnn Tensors associated with dense Tensors in the conditions above but to sparse Tensors here?

Reply:
**clone_obey_contract** in dense tensor step seems will check tensor's layout which will clone a tensor having same layout wth variable , but MKLDNN tensor has not layout attribute, so associated with sparse tensor here.

Reply:
Would it be clearer to separate these? Also updating the doc above would make these choices clearer.